### PR TITLE
Add deep mode flag on resource schema

### PR DESCRIPTION
**Pull request by eliecharra**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | yes
| 🚀 New feature?   | no
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #1068 
| ❓ Documentation  | no

## Description

This add a new component to be able to flag some resource types to be used for general purpose.

### CODE REVIEW

**Review by moadibfr — DISMISSED** [review]
LGTM :+1:

File: pkg/resource/aws/aws_api_gateway_rest_api.go [hunk 1]
```diff
@@ -1,3 +1,9 @@
 package aws
 
+import "github.com/cloudskiff/driftctl/pkg/resource"
+
 const AwsApiGatewayRestApiResourceType = "aws_api_gateway_rest_api"
+
+func initAwsApiGatewayRestApiMetaData(resourceSchemaRepository resource.SchemaRepositoryInterface) {
+	resourceSchemaRepository.SetFlags(AwsApiGatewayRestApiResourceType, resource.FlagDeepMode)
```
Comment (wbeuil) [inline pkg/resource/aws/aws_api_gateway_rest_api.go:8]: Can you remove this one? It's not `deep-mode` compatible.

File: pkg/resource/schemas.go [hunk 2]
```diff
@@ -106,6 +122,17 @@ func (r *SchemaRepository) Init(providerName, providerVersion string, schema map
 	return nil
 }
 
+func (r SchemaRepository) SetFlags(typ string, flags ...Flags) {
+	metadata, exist := r.GetSchema(typ)
+	if !exist {
+		logrus.WithFields(logrus.Fields{"type": typ}).Warning("Unable to set metadata, no schema found")
```
Comment (wbeuil) [inline pkg/resource/schemas.go:128]: `Unable to set flags ...`

**Review by wbeuil — CHANGES_REQUESTED** [review]

**Review by sundowndev — DISMISSED** [review]

**Review by wbeuil — APPROVED** [review]
Go!

File: pkg/resource/schemas.go [hunk 3]
```diff
@@ -106,6 +122,17 @@ func (r *SchemaRepository) Init(providerName, providerVersion string, schema map
 	return nil
 }
 
+func (r SchemaRepository) SetFlags(typ string, flags ...Flags) {
```
Comment (sundowndev) [inline pkg/resource/schemas.go:125]: This is minor but you could have used the `resource.ResourceType` type here

Comment (eliecharra) [reply pkg/resource/schemas.go:125]: Oops sorry I clicked on merge by the same time ...

**Pull request merged**